### PR TITLE
Prerelease/2.0.0 rc2

### LINKED ISSUE #468: system.number is 0
`api.query.system.number` returns 0, should be solved in newer updates

### PR DESCRIPTION
**Pull request by jordy25519**
## General
- Change substrate deps to build from `cennznet/substrate` (solves block propagation latency issue)
- Removes grandpa voter remote restart command

## ERC20peg
- Set_erc20_meta allows setting ERC20 asset meta directly

## NFT
- TokenLocks now has reason (makes it easier to find listingId from tokenId)

closes #468 #457

### CODE REVIEW

**Review by KarishmaBothara — APPROVED** [review]

File: crml/nft/src/types.rs [hunk 1]
```diff
@@ -341,6 +348,17 @@ pub type TokenCount = SerialNumber;
 /// Global unique token identifier
 pub type TokenId = (CollectionId, SeriesId, SerialNumber);
 
+// A value placed in storage that represents the current version of the NFT storage. This value
```
Comment (JasonTulp) [inline crml/nft/src/types.rs:351]: This is clever, I like the intent

File: ethy-gadget/src/keystore.rs [hunk 2]
```diff
@@ -85,6 +85,7 @@ impl EthyKeystore {
 	/// Use the `public` key to verify that `sig` is a valid signature for `message`.
 	///
 	/// Return `true` if the signature is authentic, `false` otherwise.
+	#[allow(dead_code)]
```
Comment (JasonTulp) [inline ethy-gadget/src/keystore.rs:88]: Why is this necessary?

**Review by JasonTulp — APPROVED** [review]
lgtm! 🙂

File: crml/nft/src/types.rs (hunk 1 above)
Comment (jordy25519) [reply crml/nft/src/types.rs:351]: this reminds me I needed to update the storage version after updating

File: ethy-gadget/src/keystore.rs (hunk 2 above)
Comment (jordy25519) [reply ethy-gadget/src/keystore.rs:88]: suppresses the compiler warning about the function being unused (more OCD than anything important)
